src/libstore/filetransfer.cc

```diff
@@ -815,7 +815,7 @@ struct curlFileTransfer : public FileTransfer
                 S3Helper s3Helper(profile, region, scheme, endpoint);
 
                 // FIXME: implement ETag
-                auto s3Res = s3Helper.getObject(parsed.bucket, parsed.key);
+                auto s3Res = s3Helper.getObject(parsed.bucket, concatStringsSep("/", parsed.key));
```

Comment:
guessing this should be `encodePath`

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html they recommend avoiding `%` because "things maybe happen" lol. but it is allowed.

but because we decoded the percents before, I think it is on us to re-encode them.